File: Leetcode 1801-1900/1861/1861.java
// Leetcode 1861: Rotating the Box
// https://leetcode.com/problems/rotating-the-box/
// Solved on 13th of November, 2025
class Solution {
    /**
     * Rotates a given box grid 90 degrees clockwise and simulates the effect of gravity.
     * Stones ('#') fall down until they hit an obstacle ('*') or another stone.
     *
     * @param boxGrid The input 2D character array representing the box.
     * @return A new 2D character array representing the rotated and settled box.
     */
    public char[][] rotateTheBox(char[][] boxGrid) {
        int m = boxGrid.length;
        int n = boxGrid[0].length;
        for (int i = 0; i < m; i++) {
            int empty = n - 1;
            for (int j = n - 1; j >= 0; j--) {
                char c = boxGrid[i][j];
                if (c == '*') {
                    empty = j - 1;
                } else if (c == '#') {
                    if (empty != j) {
                        boxGrid[i][empty] = '#';
                        boxGrid[i][j] = '.';
                    }
                    empty--;
                }
            }
        }
        char[][] result = new char[n][m];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                result[i][j] = '.';
            }
        }
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                result[j][m - 1 - i] = boxGrid[i][j];
            }
        }
        return result;
    }
}

File: Leetcode 1801-1900/1861/1861.py
# Leetcode 1861: Rotating the Box
# https://leetcode.com/problems/rotating-the-box/
# Solved on 13th of November, 2025
class Solution:
    def rotateTheBox(self, boxGrid: list[list[str]]) -> list[list[str]]:
        """
        Rotates the given box grid 90 degrees clockwise and applies gravity.

        Args:
            boxGrid: A list of lists of strings representing the box.
        Returns:
            A list of lists of strings representing the rotated and gravitated box.
        """
        m = len(boxGrid)
        n = len(boxGrid[0])
        result = [['.'] * m for _ in range(n)]

        for i in range(m):
            targetCol = m - 1 - i
            writeRow = n - 1
            for j in range(n - 1, -1, -1):
                cell = boxGrid[i][j]
                if cell == '*':
                    result[j][targetCol] = '*'
                    writeRow = j - 1
                elif cell == '#':
                    result[writeRow][targetCol] = '#'
                    writeRow -= 1

        return result